api.md

```diff
@@ -469,7 +535,66 @@ CREATE INDEX ON conditions(time, location) USING brin
 ```
 
 ---
+## detach_data_node() [](detach_data_node)
 
+Detach a data node from a hypertable or from all hypertables.
+
+This function is typically used if a data node should no longer be
+used. Then it need to be removed from all hypertables that use
+it. Another reason for detaching a data node from a hyper table is
+that you want to have fewer
+
+#### Required Arguments [](detach_data_node-required-arguments)
+
+| Name              | Description                                   |
+| `node_name`       | Local name of data node to attach             |
+
+##### Parameter Description
+
+* `node_name` - Local name of the data node to attach to the
+  hypertable.
+
+#### Optional Arguments [](detach_data_node-optional-arguments)
+
+| Name         | Description                            |
+| `hypertable` | Name of hypertable to detach node from |
+| `force`      | Force remove the data node             |
+
+##### Parameter Description
+
+* `hypertable` - The name of the hypertable where the data node should
+  be attached. If NULL, the data node will be detached from all
+  hypertables.
+
+* `if_not_attached` - Do not generate an error if the data node is
+  already attached to the hypertable. A notice will be printed that
+  the data node is attached.
+
+#### Returns
+
+| Column               | Description                              |
+| `hypertable_id`      | Hypertable id of the modified hypertable |
+| `node_hypertable_id` | Hypertable id on the remote data node    |
+| `node_name`          | Local name of the attached data node     |
+
+#### Errors
+
+If the data node contains chunks that are not replicated on other data
+nodes, and error will be given. To detach a data node chunks on that
+data node should be replicated on other data nodes.
```

Comment:
Alternatively, you can force the removal of a data node using the `force` parameter, but do note that this will result in data loss.

Reply:
You cannot force if that would mean a data loss. Forcing is for detaching even though that would leave the table with under-replicated chunks. This should probably be clarified. Basically, I think we should add in the beginning something like: "Detaching a data node is under no circumstances possible if that would mean a data loss for the hypertable. Nor is it possible to detach a data node, unless forced,  if that would mean that the distributed hypertable would end up with under-replicated chunks. The only safe way to detach a data node, however, is to first safely delete any data on it or replicate it to another data node. A common pattern for detaching a data node is also to first block new chunks on it, and then let data retention (if such a policy exists) take care of deleting any remaining data on the data node before detaching it."

Reply:
I updated the description, but did not add the last part since we then have to have descriptions of how to block new chunks on the data node or how to delete remaining data on the data node before removing it. I would prefer doing this in a separate PR.

Reply:
Adjusted!